### PR TITLE
crosswalk-embargo

### LINKED ISSUE #519: Edit Item / Embargo - metadata `local.embargo.termslift`
Original issue: https://github.com/ufal/clarin-dspace/issues/1054
**Excel Notes:** `Embargo value will be loaded from the metadata _local.embargo.termslift_ exposed by the datacite_openaire.xsl crosswalk`

2024/Apr/18:
- When the embargo is set up the dc.date.available is not removed
- v5 - termslift is assigned only for two Items: http://hdl.handle.net/11234/1-3498 and http://hdl.handle.net/11372/LRT-2729
- termslift metadata could be used like this:
```
#### Embargo Settings ####
# DC metadata field to hold the user-supplied embargo terms
embargo.field.terms = local.embargo.termslift

# DC metadata field to hold computed "lift date" of embargo
embargo.field.lift = local.embargo.termslift 
```

### PR DESCRIPTION
**Pull request by Paurikova2**
| Phases            | MP | MM  |  MB  | MR |   JM | Total  |
|-----------------|----:|----:|----:|-----:|-----:|-------:|
| ETA                  |  0  |  0  |    0 |     0 |      0 |        0 |
| Developing      |  16  |  0  |    0 |    0 |      0 |         0 |
| Review             |  0  |  0  |    0 |    0 |      0 |         0 |
| Total                |   -  |   -  |   -   |  -    |   -    |         0 |
| ETA est.             |      |      |       |       |         |         0 |
| ETA cust.           |   -  |   -  |   -  |   -   |   -     |        0 |
## Problem description
In v5 there is embargo as metadata termslift. In v7 the embargo is managing by resource policy. Add this info to crosswalk.

### CODE REVIEW

**Comment by Paurikova2** [comment]
@milanmajchrak What's about when resource policy has end_date too. Shouldn't be this date as Available for crosswalk??

File: dspace-oai/src/main/java/org/dspace/utils/SpecialItemService.java [hunk 1]
```diff
@@ -267,6 +281,159 @@ public static Node getAuthor(String mdValue) {
...
+
+    /**
+     * Finds an item in DSpace using the provided handle and metadata field.
+     *
+     * @param context The DSpace context
+     * @param metadataField The metadata field used for item search
+     * @param handle The handle (identifier) of the item.
+     * @return The Item object, or null if no item is found.
+     */
+    private static Item findItemByHandle(Context context, MetadataField metadataField, String handle) {
```
Comment (milanmajchrak) [inline dspace-oai/src/main/java/org/dspace/utils/SpecialItemService.java]: Duplicity of this: https://github.com/dataquest-dev/DSpace/blob/dtq-dev/dspace-api/src/main/java/org/dspace/content/service/clarin/ClarinItemService.java#L46

**Review by milanmajchrak — CHANGES_REQUESTED** [review]

File: dspace-oai/src/main/java/org/dspace/xoai/services/impl/resources/functions/StringXSLFunction.java [hunk 2]
```diff
@@ -74,7 +74,6 @@ final public XdmValue call(XdmValue[] xdmValues) throws SaxonApiException {
             log.warn("Empty value in call of function of StringXslFunction type");
             val = "";
         }
-
```
Comment (milanmajchrak) [inline dspace-oai/src/main/java/org/dspace/xoai/services/impl/resources/functions/StringXSLFunction.java:77]: Why?

**Review by milanmajchrak — CHANGES_REQUESTED** [review]

File: dspace-oai/src/main/java/org/dspace/xoai/services/impl/resources/functions/StringXSLFunction.java (hunk 2 above)
Comment (Paurikova2) [reply dspace-oai/src/main/java/org/dspace/xoai/services/impl/resources/functions/StringXSLFunction.java:77]: It looks better :)

Comment (Paurikova2) [reply dspace-oai/src/main/java/org/dspace/xoai/services/impl/resources/functions/StringXSLFunction.java:77]: I added line but I got checkstyle violation so I had to remove it again....

**Pull request merged**